src/javascript/ContentEditor/validation/validate.js

```diff
@@ -70,6 +70,46 @@ const colorFieldValidation = (values, field) => {
     return undefined;
 };
 
+const rangeFieldValidation = (values, field) => {
+    const error = 'invalidRange';
+    const supportedTypes = ['LONG', 'DECIMAL', 'DOUBLE'];
+    if (!supportedTypes.includes(field.requiredType) || !values[field.name]) {
+        return;
+    }
+
+    const fieldValues = Array.isArray(values[field.name]) ? values[field.name] : [values[field.name]];
+    // If one value is invalid, error !
+    if (fieldValues.some(value => isNaN(value))) {
+        return error;
+    }
+
+    if (!field.valueConstraints || field.valueConstraints.length === 0) {
+        return;
+    }
+
+    return field.valueConstraints
+        .map(constraint => {
+            try {
+                const {lowerBoundary, disableLowerBoundary, upperBoundary, disableUpperBoundary} = extractRangeConstraints(constraint.value.string);
+
+                return fieldValues
+                    .every(fieldValue => {
+                        const fieldValueAsNumber = Number(fieldValue);
+                        // Lower boundary Check
+                        const lowerBoundaryOk = disableLowerBoundary ? fieldValueAsNumber > Number(lowerBoundary) : fieldValueAsNumber >= Number(lowerBoundary);
+                        // Upper boundary Check
+                        const upperBoundaryOk = disableUpperBoundary ? fieldValueAsNumber < Number(upperBoundary) : fieldValueAsNumber <= Number(upperBoundary);
+                        return lowerBoundaryOk && upperBoundaryOk;
```

Comment:
Minor optimisation: no need to perform an upper boundary check if the lower boundary check failed:
```suggestion
                        return 
                        // Lower boundary Check
                        (disableLowerBoundary ? fieldValueAsNumber > Number(lowerBoundary) : fieldValueAsNumber >= Number(lowerBoundary)) && 
                        // Upper boundary Check
                        (disableUpperBoundary ? fieldValueAsNumber < Number(upperBoundary) : fieldValueAsNumber <= Number(upperBoundary));
```